app/test_main.py

```diff
@@ -1 +1,62 @@
-# write your code here
+import pytest
+
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture
+def mocked_validation() -> mock.Mock:
+    with mock.patch("app.main.valid_google_url") as mocked_validation:
+        yield mocked_validation
+
+
+@pytest.fixture
+def mocked_connection() -> mock.Mock:
+    with mock.patch("app.main.has_internet_connection") as mocked_connection:
+        yield mocked_connection
+
+
+@pytest.mark.parametrize(
+    "validation_return,connection_return,access_return",
+    [
+        pytest.param(True,
+                     True,
+                     "Accessible",
+                     id="when validation and connection are true, "
+                        "page should be accessible"),
+
+        pytest.param(True,
+                     False,
+                     "Not accessible",
+                     id="when validation is true and no connection, "
+                        "page should not be accessible"),
+
+        pytest.param(False,
+                     True,
+                     "Not accessible",
+                     id="when no validation and connection is true, "
+                        "page should not be accessible"),
+
+        pytest.param(False,
+                     False,
+                     "Not accessible",
+                     id="when no validation and no connection "
+                        "are true, page should not be accessible")
+    ]
+)
+def test_can_access_google_page(
+        mocked_validation: mock.Mock,
+        mocked_connection: mock.Mock,
+        validation_return: bool,
+        connection_return: bool,
+        access_return: str
+) -> None:
+
+    mocked_validation.return_value = validation_return
+    mocked_connection.return_value = connection_return
+    assert can_access_google_page("") == access_return
+    if mocked_connection():
```

Comment:
here `has_internet_connection` will be called always and `valid_google_url` will be called only if `has_internet_connection` returned True